### PR TITLE
Client rewrite

### PR DESCRIPTION
**Pull request by AphonicChaos**
# TL;DR
This PR is a completely rewrite of the client code, though I anticipate that I'll back-port the authentication code from the other branch. It's currently a draft, which is why the `warwomb-client-old` directory exists, rather than being deleted outright. 

# TODO
- [x] add a "surface" behind the left and right player token displays, as well as the "action" area on top so text and other content is easier to see
- [ ] provide some means of collapsing/subduing those sections so they aren't distracting during play
- [ ] back-port the Auth0 authentication code
- [ ] re-implement draggable tokens
- [ ] separate display  into two layers:
    - "hud" layer where ui elements that aren't meant to scale or be dragged around can go
    - "interactive" layer where the map itself can be zoomed and panned, scaling elements with it, and elements can be marked as movable, having menus, etc
- [ ] allow the map to be scaled/rotated like before

Note that adding the layer is all I intend to do before merging -- making them interactive is a different story

# Screenshots
Obviously, from the after you can tell this is still work-in-progress.
## Before
2023-04-16-100841_2917x1476_scrot

## After
2023-06-10-201736_1440x1177_scrot

### CODE REVIEW

**Comment by AphonicChaos** [comment]
After drawing the player areas onto a surface, text is legible, but it's also even more apparent that I need to work on splitting those "layers", and likely move away from CSS grid to give me more control over adding and such of the grid items (which is why everything looks so squished in the screenshot).
